src/Client.ts

```diff
@@ -0,0 +1,1232 @@
+    wait?: boolean,
+  ): Promise<Message | void> {
+    let embeds: embed.Embed[] | undefined;
+    if (data.embeds) {
+      embeds = data.embeds.map((embed) => unparseEmbed(embed));
+    }
+
+    let convertedData: webhook.ExecuteBody = {
+      content: data.content,
+      tts: data.tts,
+      embeds,
+      allowed_mentions: data.allowedMentions,
+    };
+
+    if (data.file) {
+      convertedData = {
+        file: data.file,
+        payload_json: JSON.stringify(convertedData),
+      };
+    }
+
+    const res = await this.rest.executeWebhook(
+      webhookId,
+      token,
+      convertedData,
+      {
+        wait,
+      },
+    );
+
+    if (wait) {
+      // TODO: cast shouldn't be unnecessary but ts complains
+      return new Message(this, res as message.Message);
+    }
+  }
+
+  /** Requests guild members. These then will be received through the guildMembersChunk event. */
+  requestGuildMembers(
+    shardNumber: number,
+    guildIds: Snowflake | Snowflake[],
+    options: {
+      query?: string;
+      limit: number;
+      presences?: boolean;
+      userIds?: Snowflake | Snowflake[];
+      nonce?: string;
+    },
+  ): void {
+    this.gateway.guildRequestMember(shardNumber, {
+      guild_id: guildIds,
+      query: options.query,
+      limit: options.limit,
+      presences: options.presences,
+      user_ids: options.userIds,
+      nonce: options.nonce,
+    });
+  }
+
+  /** Updates the current user's status. */
+  statusUpdate(shardNumber: number, data: {
+    since: number | null;
+    game: Activity | null;
+    status: presence.ActiveStatus;
+    afk: boolean;
+  }): void {
+    this.gateway.statusUpdate(shardNumber, {
+      since: data.since,
+      game: data.game && unparseActivity(data.game),
+      status: data.status,
+      afk: data.afk,
+    });
+  }
+
+  // utils
+
+  newChannelSwitch(data: channel.Channel): Channel {
+    switch (data.type) {
+      case 0:
+        return new TextChannel(this, data);
+      case 1:
+        return new DMChannel(this, data);
+      case 2:
+        return new VoiceChannel(this, data);
+      case 3:
+        return new GroupDMChannel(this, data);
+      case 4:
+        return new CategoryChannel(this, data);
+      case 5:
+        return new NewsChannel(this, data);
+      case 6:
+        return new StoreChannel(this, data);
+    }
+  }
+
+  async sendMessage(
+    channelId: Snowflake,
+    data: SendMessageOptions,
+  ): Promise<Message> {
+    let embed;
+    if (data.embed) {
+      embed = unparseEmbed(data.embed);
+    }
+
+    let convertedData: message.BaseCreate = {
+      content: data.content,
+      tts: data.tts,
+      embed,
+      allowed_mentions: data.allowedMentions,
+    };
+
+    if (data.file) {
+      convertedData = {
+        file: data.file,
+        payload_json: JSON.stringify(convertedData),
+      };
+    }
+
+    const message = await this.rest.createMessage(
+      channelId,
+      convertedData as message.Create,
```

Comment:
Unnecessary cast? Do `const convertedData: message.Create = convertedData;` instead, if the type needs to be massaged first — though if it does, there might be something wrong here, or the type of `convertedData` is not specific enough.